addons/controllers/addon_controller.go

```diff
@@ -115,12 +116,6 @@ func (r *AddonReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ct
 		return ctrl.Result{}, err
 	}
 
-	tkrName := util.GetTKRNameForCluster(ctx, r.Client, cluster)
```

Comment:
I am wondering if we should keep this because the predicate only checks for the existence of a label key and not the actual existence of a TKR object and corresponding BOM configmap. Note that BOM configmap is only needed for addon_controller and not clusterbootstrap_controller.go

Without the check reconcileNormal of addon_controller will return an error and keep re-queuing if BOM is not found.

Reply:
This function ```GetTKRNameForCluster()``` was solely checking the label previously, it wasn't checking if the tkr object existed. Please note that ```GetTKRForCluster()``` was not used at all.

Reply:
You are right, I got confused by the similarities in the function names. Checking for BOM would be great in the addons_controller.

Reply:
@vijaykatam BOM is already checked in the ```ReconcileNormal()``` logic```bom, err := util.GetBOMForCluster(ctx,r.Client, cluster)```.
And as far I can tell, ```ReconcileDelete()``` wouldn't need the BOM to exist.